### PR TITLE
MODQM-127 MARC record does NOT open after saving an invalid field

### PR DESCRIPTION
**Pull request by oleksandr-dekin**
## Purpose
Added validation for marc tags, checking on the number symbols

## Approach
Extended validation service by new rule for marc tags. We should exclude symbolic value in tags and use only with next format: `001-999`

## Learning
https://issues.folio.org/browse/MODQM-127

### CODE REVIEW

File: src/main/java/org/folio/qm/service/impl/MarcRecordsServiceImpl.java [hunk 1]
```diff
@@ -71,6 +71,7 @@ public QuickMarc findByInstanceId(UUID instanceId) {
   @Override
   public void updateById(UUID instanceId, QuickMarc quickMarc) {
     validationService.validateIdsMatch(quickMarc, instanceId);
+    validationService.validateTags(quickMarc);
```
Comment (psmagin) [inline src/main/java/org/folio/qm/service/impl/MarcRecordsServiceImpl.java:74]: We have a pattern for tags in API spec. To validate it the simplest way is to add dependency 
```
<dependency>
      <groupId>org.hibernate.validator</groupId>
      <artifactId>hibernate-validator</artifactId>
    </dependency>
```
and add a related Exception handler to ConrollerAdvice class.

File: src/main/java/org/folio/qm/service/impl/MarcRecordsServiceImpl.java [hunk 2]
```diff
@@ -71,6 +71,7 @@ public QuickMarc findByInstanceId(UUID instanceId) {
   @Override
   public void updateById(UUID instanceId, QuickMarc quickMarc) {
```
Comment (psmagin) [inline src/main/java/org/folio/qm/service/impl/MarcRecordsServiceImpl.java:72]: And what about creating a new record? We should validate it also.

File: src/main/java/org/folio/qm/service/impl/MarcRecordsServiceImpl.java (hunk 1 above)
Comment (oleksandr-dekin) [reply src/main/java/org/folio/qm/service/impl/MarcRecordsServiceImpl.java]: Added dependency and extended ErrorHandling

File: src/main/java/org/folio/qm/service/impl/MarcRecordsServiceImpl.java (hunk 2 above)
Comment (oleksandr-dekin) [reply src/main/java/org/folio/qm/service/impl/MarcRecordsServiceImpl.java]: Yes, it is currently working when we use `hibernate-validator`

File: src/main/java/org/folio/qm/controller/ErrorHandling.java [hunk 3]
```diff
@@ -1,29 +1,32 @@
...
+import org.springframework.web.bind.MethodArgumentNotValidException;
 import org.springframework.web.bind.MissingServletRequestParameterException;
 import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.web.bind.annotation.ResponseStatus;
 import org.springframework.web.bind.annotation.RestControllerAdvice;
 import org.springframework.web.method.annotation.MethodArgumentTypeMismatchException;
 
-import org.folio.qm.exception.QuickMarcException;
-import org.folio.spring.exception.NotFoundException;
-import org.folio.tenant.domain.dto.Error;
```
Comment (psmagin) [inline src/main/java/org/folio/qm/controller/ErrorHandling.java:22]: Please check your import order in IDE. The default order on folio is:
1. static not org.folio.*
2. blank line
2. static   org.folio.*
3. blank line
4. java.* and javax.*
5. blank line
6. not org.folio.*
7. blank line
8. org.folio.*

File: src/main/java/org/folio/qm/controller/ErrorHandling.java [hunk 4]
```diff
@@ -1,29 +1,32 @@
...
-
 @RestControllerAdvice
 public class ErrorHandling {
 
+  public static final String PARAMETER = "Parameter '";
```
Comment (psmagin) [inline src/main/java/org/folio/qm/controller/ErrorHandling.java:28]: It's not clear what this constant for.
I suggest having constants with error message patterns. For example:
```suggestion
  private static final String TYPE_MISMATCH_MSG_PATTERN = "Parameter '%s' is invalid";
```

File: src/main/java/org/folio/qm/controller/ErrorHandling.java [hunk 5]
```diff
@@ -47,34 +62,38 @@ public Error handleConverterException(QuickMarcException e, HttpServletResponse
...
+    var message = PARAMETER + e.getParameterName() + "' is required";
+    return buildBadRequestResponse(message);
   }
 
   @ExceptionHandler(MethodArgumentTypeMismatchException.class)
-  public Error handleGlobalException(MethodArgumentTypeMismatchException e, HttpServletResponse response) {
-    var message = "Parameter '" + e.getParameter().getParameterName() + "' is invalid";
-    return buildBadRequestResponse(response, message);
+  @ResponseStatus(HttpStatus.BAD_REQUEST)
+  public Error handleGlobalException(MethodArgumentTypeMismatchException e) {
```
Comment (psmagin) [inline src/main/java/org/folio/qm/controller/ErrorHandling.java:79]: It's not global exception. Please rename method

**Review by psmagin — APPROVED** [review]

**Review by natali-zaitseva — APPROVED** [review]

**Pull request merged**